scripts/git/update_tf_submodule.sh

```diff
@@ -33,30 +33,28 @@ set -e
 set -o pipefail
 
 export UPSTREAM_REMOTE="${UPSTREAM_REMOTE:-upstream}"
-TENSORFLOW_COMMIT="${1:-REMOTE}"
+TENSORFLOW_COMMIT="${1:-LATEST_MATCH}"
```

Comment:
So the default behaviors are a little different depending on the entry point used?
* GitHub Actions (`update_tf.yml`): Calls `update_to_llvm_syncpoint.py` directly -> `INTEGRATE_COMMIT` (remains fixed until the next LLVM integrate)
* Build cop (`update_tf_submodule.sh`): Calls with `LATEST_MATCH` (should realistically match `REMOTE_HEAD_COMMIT`, unless something weird happens with Copybara or it gets run in the brief window between Copybara exports)

Makes sense to me, just checking that those are the intended defaults.

Reply:
Hmmm no that actually wasn't intentional to have the diff. I kept changing my mind which was the better default :-D I think for these it makes sense to use LATEST_MATCH. At some point I was thinking about triggering this directly on an LLVM change, in which case it might make sense to look for the integrate commit itself.

Reply:
Switched to make LATEST_MATCH the default for the python script as well

Reply:
Also note that we actually currently have a mismatch with the TF LLVM version and the IREE LLVM version on main, so the case for `LATEST_MATCH` is not as unlikely as it seems, I think :stuck_out_tongue_winking_eye: